webapp/src/services/api/study.ts

```diff
@@ -166,8 +166,15 @@ export const unarchiveStudy = async (sid: string): Promise<void> => {
   await client.put(`/v1/studies/${sid}/unarchive`);
 };
 
-export const upgradeStudy = async (sid: string): Promise<void> => {
-  await client.put(`/v1/studies/${sid}/upgrade`);
+export const upgradeStudy = async (
+  studyId: string,
+  targetVersion: string
+): Promise<void> => {
+  await client.put(
+    `/v1/studies/${studyId}/upgrade?target_version=${encodeURIComponent(
```

Comment:
`encodeURIComponent` is necessary for version?

Reply:
you are right, it might not always be strictly necessary, it's often a good idea to encode URI params to be safe. As you like I can remove it

Reply:
I agree that it is not necessary to validate the values since they are provided by the back office and guaranteed to be numbers (in other words, the values do not come from user input). However, it is important to note that the value can be generated from an R script using the API. Therefore, my question is whether you ensure that the value matches a certain regex, for instance, `"\d+"`?

Reply:
I am not sure to understand, this is not validation, just encoding for URI parameters when we send the update request:
this ensure to have proper encoding for URI components (encode spaces or specials chars etc..), also useful for security aspects as it prevent potential injections by manipulating the payload request, see:
[https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent](url)